utils.ts

```diff
@@ -241,3 +237,38 @@ export const mergeResponses = (
 
   return mergedResponse
 }
+
+/**
+ * Makes a geocoder request by first checking a potential redis store for a cached
+ * response. If a response is fetched, stores response in redis store.
+ * @param redisClient     Redis client already connected
+ * @param requestMethod   Geocoder Request Method
+ * @param args            Args for Geocoder request method
+ * @returns               FeatureCollection either from cache or live
+ */
+export const cachedGeocoderRequest = async (
+  geocoder: Record<string, (q: AnyGeocoderQuery) => Promise<FeatureCollection>>,
+  requestMethod: string,
+  args: AnyGeocoderQuery,
+  redisClient: RedisClientType | null
+): Promise<FeatureCollection> => {
+  const { focusPoint, text } = args
+  if (!text) return { features: [], type: 'FeatureCollection' }
+  const stringifiedFocusPoint = `${focusPoint?.lat}.${focusPoint?.lon}`
+  const redisKey = text + stringifiedFocusPoint
+
+  if (redisClient) {
+    const cachedResponse = await redisClient.get(redisKey)
+    if (cachedResponse) {
+      return JSON.parse(cachedResponse)
+    }
+  }
+  const onlineResponse = await geocoder[requestMethod](args)
+  // If we are at this point and have a redis object we know there
+  // was no entry in the cache
+  if (redisClient) {
+    redisClient.set(redisKey, JSON.stringify(onlineResponse))
```

Comment:
Good catch thank you! Would be annoying to break the entire thing because of a bad key or something. Refactored in d8598b93ef1d8cfc2e66faffd38be81c9a9e442c!